### PR TITLE
Add fight cave plugin

### PR DESCRIPTION
**Pull request by devinfrench**
preview

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java [hunk 1]
```diff
@@ -71,13 +71,16 @@
...
+
+	PRAYER_PROTECT_FROM_MAGIC(WidgetID.PRAYER_GROUP_ID, WidgetID.Prayer.PROTECT_FROM_MAGIC),
+	PRAYER_PROTECT_FROM_MISSILES(WidgetID.PRAYER_GROUP_ID, WidgetID.Prayer.PROTECT_FROM_MISSILES);
 
 
 	private final int groupId;
 	private final int childId;
 
-	private WidgetInfo(int groupId, int childId)
+	WidgetInfo(int groupId, int childId)
```
Comment (Adam-) [inline runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java]: ?

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/JadAttack.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+import net.runelite.api.Client;
+import net.runelite.api.Prayer;
+import net.runelite.api.widgets.Widget;
+import net.runelite.api.widgets.WidgetInfo;
+
+public enum JadAttack
+{
+
+	MAGIC(2656, Prayer.PROTECT_FROM_MAGIC, WidgetInfo.PRAYER_PROTECT_FROM_MAGIC),
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/JadAttack.java]: Add these to the AnimationID class

File: runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+		return null;
+	}
+
+	private Rectangle getViewportBounds()
+	{
+		Widget viewport = getViewportWidget();
+		return viewport != null ? viewport.getBounds() : null;
+	}
+
+	private Widget getViewportWidget()
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/fightcave/FightCaveOverlay.java]: This logic should probably be moved to somewhere more central. Maybe into the client mixin

File: runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java (hunk 1 above)
Comment (devinfrench) [reply runelite-api/src/main/java/net/runelite/api/widgets/WidgetInfo.java]: All enum constructors are implicitly private, just like all interface methods are implicitly public.

**Comment by Adam-** [comment]
Thanks

**Pull request merged**